### PR TITLE
Use a passthrough filter if Anonymization disabled

### PR DESCRIPTION
**Pull request by Dohbedoh**
* Remove the use of `Optional<ContentFilter>` passed in parameters through the API. Use a `NoneFilter` implementation instead that simply pass through the value. This should simplify the readability, also avoid NPE and remove the need for null checks in some part of the code.
* Deprecated the existing methods `SupportPlugin#getContentFilter` / `SupportPlugin#getContentFilter(ensureLoaded)` in favor of `SupportPlugin#getDefaultContentFilter` / `SupportPlugin#getDefaultContentFilter(ensureLoaded)` and make sure that `ensureLoaded` also reloads the mappings.
* Deprecate `ContentFilter#ensureLoaded` that does not seem useful in favor of just using the existing `ContentFilter#reload`. 
* Also fixed checkstyle in the classes part of the fix

### Testing done

Tested on a test instance with thousands of items and anonymization off then on.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by jglick — APPROVED** [review]
Looks simpler.

File: src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java [hunk 1]
```diff
@@ -381,26 +381,20 @@ public static void writeBundle(
...
+                        appendManifestContents(manifest, errorWriter, components, componentConsumer, filter);
                 LOGGER.log(
                         Level.FINE,
                         "Took " + (System.currentTimeMillis() - startTime) + "ms to process all components");
                 contents.add(new UnfilteredStringContent("manifest.md", manifest.toString()));
 
-                Optional<FilteredOutputStream> maybeFilteredOut =
-                        maybeFilter.map(filter -> new FilteredOutputStream(binaryOut, filter));
-                OutputStream textOut =
-                        maybeFilteredOut.map(OutputStream.class::cast).orElse(binaryOut);
```
Comment (dwnusbaum) [inline src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java:403]: @Dohbedoh For what it's worth, while debugging issues with a log file using a support bundle, after a bit of confusion I think that this PR had the side effect of converting Windows line endings in files in the support bundle to Unix line endings, even when anonymization is disabled, at least in some cases.

I think the key difference is that in the old code, if anonymization was disabled, we just passed everything through via binary streams with no processing. With the new code, even if anonymization is disabled, we pass everything through `FilteredOutputStream`, which decodes the binary data into text, applies the filter, and then re-encodes the text, and I guess somewhere along the way line endings get normalized. Beyond the line endings issue though, I suspect performance might also be worse due to the increased processing overhead when anonymization is disabled.

File: src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java [hunk 2]
```diff
@@ -122,8 +123,8 @@ protected void writeTo(OutputStream os) throws IOException {
         }
     }
 
-    protected void writeTo(OutputStream os, ContentFilter filter) throws IOException {
-        if (isBinary || filter == null) {
```
Comment (dwnusbaum) [inline src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java:126]: Dropping this condition was actually the reason for https://github.com/jenkinsci/support-core-plugin/pull/502/files#r1491768009, see https://github.com/jenkinsci/support-core-plugin/pull/509.

File: src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java (hunk 1 above)
Comment (dwnusbaum) [reply src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java:403]: The specific issue was actually with `BaseFileContent`, see https://github.com/jenkinsci/support-core-plugin/pull/502/files#r1492687323 and https://github.com/jenkinsci/support-core-plugin/pull/509.

File: src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java (hunk 2 above)
Comment (dwnusbaum) [reply src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java:126]: Well, only partially. Previously we would have gone through the `writeTo(OutputStream)` overload above instead. If this was the only issue, then we would have noticed the lack of `return` causing most types of file content to be duplicated!

File: src/main/java/com/cloudbees/jenkins/support/filter/ContentFilter.java [hunk 3]
```diff
@@ -76,8 +83,8 @@ default void reload() {}
      * @param text the text to filter
      * @return the text filtered if it is not empty and the filter is not null
      */
-    static String filter(@CheckForNull ContentFilter filter, @CheckForNull String text) {
-        if (filter != null && StringUtils.isNotEmpty(text)) {
+    static String filter(@NonNull ContentFilter filter, @CheckForNull String text) {
+        if (StringUtils.isNotEmpty(text)) {
```
Comment (jglick) [inline src/main/java/com/cloudbees/jenkins/support/filter/ContentFilter.java:87]: https://github.com/jenkinsci/support-core-plugin/pull/522#discussion_r1521811757

**Pull request merged**